compute/metadata/metadata.go

```diff
@@ -304,9 +306,21 @@ func (c *Client) getETag(suffix string) (value, etag string, err error) {
 	}
 	req.Header.Set("Metadata-Flavor", "Google")
 	req.Header.Set("User-Agent", userAgent)
-	res, err := c.hc.Do(req)
-	if err != nil {
-		return "", "", err
+	var res *http.Response
+	retryer := newRetryer()
+	for {
+		var err error
+		res, err = c.hc.Do(req)
+		if err != nil {
```

Comment:
Nit: You could invert the `if` here. Matter of style preference I think though.
```go
res, err = c.hc.Do(req)
if err == nil {
  break
}

if delay, shouldRetry := ...
```

Reply:
I like it!